packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -52,6 +53,15 @@ export const runCommand = async (
 		doneText: opts.doneText,
 		promise() {
 			const [executable, ...args] = command;
+			// Don't send metrics data on any wrangler commands if telemetry is disabled in C3
+			// If telemetry is disabled separately for wrangler, wrangler will handle that
+			if (args[0] === "wrangler") {
+				const metrics = readMetricsConfig();
+				if (metrics.c3permission?.enabled === false) {
+					opts.env ??= {};
+					opts.env["WRANGLER_SEND_METRICS"] = "false";
+				}
+			}
```

Comment:
Shall we make it an helper and pass the adjusted env to `runCommand` only on places we run the wrangler command?

Reply:
discussed + intentionally leaving this here so it catches all wrangler commands in one spot